### PR TITLE
fix(Authorization): Relax restrictions in the job controller

### PR DESCRIPTION
**Pull request by AbdulRahmanAlHamali**
This relaxes the restrictions in the job controller, by setting `getFileContents` and `collectJob` permissions to `READ`, and `cancelJob` permissions to `EXECUTE`.
Fixes https://github.com/spinnaker/spinnaker/issues/5454

### CODE REVIEW

**Comment by AbdulRahmanAlHamali** [comment]
Hey @ethanfrogers , I hope you're doing well :) 

I'm not really sure :thinking: 

Is the job controller used by anything other than the `RunJob` stage? If not, then honestly, it might not even need access control at all, since the access control will be carried out at earlier levels (gate or orca).

However, there is a string in the code saying "might modify the job". I think this is why they added these restrictions, but I don't fully understand what they're talking about

**Comment by ethanfrogers** [comment]
That's what I'm a bit confused by as well. If it were to modify a job then I'd actually expect this controller to care about cloud provider permissions instead of application permissions. In a similar vein, since collecting job details is actually reading from cloud resources, I'd expect that it would be controlled by those permissions as well.

**Comment by AbdulRahmanAlHamali** [comment]
Agreed. I'm looking at the source of those lines, and it seems to be a PR by Travis @ttomsu 

https://github.com/spinnaker/clouddriver/pull/1037/files

Some of the things were originally set to `READ`, but were replaced by `WRITE` because it was a POST request. Seems like an oversight overall.

Travis, do you happen to remember why we had these authorizations here?

**Comment by ttomsu** [comment]
Commented on each one. I don't know how RunJob is actually used in the wild, but that first method's permissions look correct to me.

**Comment by ajordens** [comment]
This also looks reasonable to me ...

**Comment by AbdulRahmanAlHamali** [comment]
@ttomsu why is it correct? I see it is collecting the job information, not writing to either of the application or the account. Unless there is something I'm missing :thinking:

**Comment by ttomsu** [comment]
I'm basing it on the API documentation right there that says "may modify the job". If that is not correct, then that line should be updated, changed to HTTP GET, and I would agree that READ is fine.

**Comment by AbdulRahmanAlHamali** [comment]
I see, would changing it to GET be easy? Or will it require some cross-service PR or something?

**Comment by ttomsu** [comment]
Sorry, I think I never actually submitted my review comments (just the top level one) from before.

As for the HTTP GET call, most of the time it's Orca and Gate calling Clouddriver, so those would the first two services I'd expect to check for calls to the `JobController`. So to answer your question, it may be an easy change or it may not be - if there is no POST body it may be pretty easy, but if it has a body it may not be so easy. I just don't know.

**Comment by AbdulRahmanAlHamali** [comment]
Thank you Travis. I did some research, and it seems the only one calling `collectJob` is orca, and only in the context of the `RunJob` stage.

I will update the PR to make it `GET`, and will open a PR in orca as well.

Side note: Was clouddriver called Kato before? If not, then what is Kato?

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/JobController.groovy [hunk 1]
```diff
@@ -39,7 +39,7 @@ class JobController {
   @Autowired
   MessageSource messageSource
 
-  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'WRITE') and hasPermission(#account, 'ACCOUNT', 'WRITE')")
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ') and hasPermission(#account, 'ACCOUNT', 'READ')")
```
[the remark this answers is not in the document]
Comment (AbdulRahmanAlHamali) [reply clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/JobController.groovy]: I think this is the culprit:
https://github.com/spinnaker/clouddriver/blob/198cd7aaed745c31c45315bbf5a89f44cbe17c8e/clouddriver-kubernetes-v1/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesJobProvider.groovy#L64

They used to delete the pod in v1, which is why it was considered a write. In v2, they don't:
https://github.com/spinnaker/clouddriver/blob/a7e766fa9e4e0ee09752c10ffeb0c3f03ff81891/clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/provider/view/KubernetesV2JobProvider.java#L56

**Comment by AbdulRahmanAlHamali** [comment]
I addressed the comments, and also opened a PR in Orca to move from `POST` to `GET`:
https://github.com/spinnaker/orca/pull/3454

**Comment by ttomsu** [comment]
Yes, Clouddriver used to be 3 separate services - Kato (mutating calls), Oort (read-only view of the cache), and Mort (polling and cache updating).

**Comment by AbdulRahmanAlHamali** [comment]
I moved the authorization logic of `collectJob` to the inside, and made it based on the instance of the JobProvider

**Comment by AbdulRahmanAlHamali** [comment]
Are we good to merge this?

**Comment by AbdulRahmanAlHamali** [comment]
BTW, do accounts have `EXECUTE` authorization in the first place? :thinking:

**Comment by AbdulRahmanAlHamali** [comment]
Can we merge this soon guys?

**Pull request merged**